src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,199 @@
+			}
+			catch
+			{
+			}
+
+			int iconId;
+			if (info != null)
+			{
+				iconId = info.Icon;
+			}
+			else
+			{
+				// try another method - using more resources (CPU and memory), as Intent is involved
+				Android.Content.Intent intent = Android.App.Application.Context.PackageManager.GetLaunchIntentForPackage(Android.App.Application.Context.PackageName);
+				Android.Content.PM.ResolveInfo resolveInfo = Android.App.Application.Context.PackageManager.ResolveActivity(intent, Android.Content.PM.PackageInfoFlags.MatchDefaultOnly);
+				iconId = resolveInfo.IconResource;
+			}
+
+			builder.SetSmallIcon(iconId);
+
+
+			// for somewhat older Android, we need to set Priority (deprecated in API26)
+
+			if (Android.OS.Build.VERSION.SdkInt < Android.OS.BuildVersionCodes.O)
+			{
+#pragma warning disable CS0618 // Type or member is obsolete
+				builder.SetPriority(0); // PRIORITY_DEFAULT
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+
+
+			// mirroring / bridging of notification
+			builder.SetLocalOnly((notification.NotificationMirroring == NotificationMirroring.Disabled));
+
+			// expiration time
+			if((notification.ExpirationTime != null) && (notification.ExpirationTime.HasValue) )
+			{
+				int seconds = (int)(notification.ExpirationTime.Value - DateTime.Now).TotalSeconds;
+				if(seconds>0)
+				{
+					builder.SetTimeoutAfter(seconds * 1000);
+				}
+
+			}
+
+
+			// action - opening app from tap on Toast (App.OnActivated)
+
+			var intentAction = Android.App.Application.Context.PackageManager.GetLaunchIntentForPackage(packageName);
+			intentAction.SetPackage(packageName);
+			// according to doc:
+			// The name must include a package prefix, for example the app com.android.contacts would use names like "com.android.contacts.ShowAll"
+			// but it works also in this way, and getting package name in background app can be difficult.
+			intentAction.PutExtra("Uno.internal.IntentType", (double)ApplicationModel.Activation.ActivationKind.ToastNotification);
+
+			string toastArgument = "";
+			var xmlDoc = new XmlDocument();
+			try
+			{
+				xmlDoc.LoadXml(notification.Content.GetXml());  // Windows.Data.Xml.Dom is almost not implemented, so we have to use System.Xml
+				var childToast = xmlDoc.GetElementsByTagName("toast");
+				if(childToast.Count == 1)
+				{
+					var childLaunch = childToast.Item(0).Attributes.GetNamedItem("launch");
+					if(childLaunch != null)
+					{
+						toastArgument = childLaunch.Value;
+					}
+				}
+			}
+			catch
+			{
+				// error parsing XML - maybe simple not 'launch' attribute
+				toastArgument = "";
+			}
+			intentAction.PutExtra("Uno.internal.ToastArgument", toastArgument);
+
+			// we should have intent.data field different for different toast arguments,
+			// else - next intent would overwrite previous.
+			var data = Android.Net.Uri.Parse("http://unotask.com/toast?arg=" + Android.Net.Uri.Encode(toastArgument));

```

Comment:
Where is this url coming from ?

Reply:
From my mind :)
It just has to be some Uri, different for different arguments. It can be "http://dummy.com/dummy?arg=".

Reply:
On after thought, let's use an explicitly invalid domain, to avoid confusion:

```suggestion
			var data = Android.Net.Uri.Parse("http://example.com/toast?arg=" + Android.Net.Uri.Encode(toastArgument));

```

Reply:
I think this is not good idea - as app developer can use example.com in his/her code, and we don't want any conflict here.

Reply:
Can I mark this conversation as 'resolved'?

Reply:
No, using a non-existing uno-specific Url is not correct here, particularly if it's something that needs to be used explicitly in the user's code. If this is not something that is exposed by the Windows APIs, we'll need to expose it somehow through the `FeatureConfiguration` classes.

Reply:
No, it is not meant to be used in user code.
Maybe "dummy.uno.com", or "not.real.uno.com"? (.com, as if Uri constructor would check for Top Level Domain validity, last part of DNS name should look like something that can exist).

Reply:
Can I get answer to question from May?
Can this dummy URL be in form of "dummy.uno.com", or "not.real.uno.com"?

Reply:
Let's use `android-toast.platform.uno` as an url that controlled by uno.